packages/core/src/types.ts

```diff
@@ -68,13 +75,220 @@ export interface Config {
+     *
+     * @type {number|undefined}
+     */
+    this.maxConnectionLifetime = 60 * 60 * 1000 // 1 hour
+
+    /**
+     * The maximum amount of time to wait to acquire a connection from the pool (to either create a new
+     * connection or borrow an existing one.
+     *
+     * **Default**: ```60000 // 1 minute```
+     *
+     * @type {number|undefined}
+     */
+    this.connectionAcquisitionTimeout = 60000 // 1 minute
+
+    /**
+     * Specify the maximum time in milliseconds transactions are allowed to retry via
+     * {@link Session#executeRead} and {@link Session#executeWrite} functions.
+     *
+     * These functions will retry the given unit of work on `ServiceUnavailable`, `SessionExpired` and transient
+     * errors with exponential backoff using initial delay of 1 second.
+     *
+     * **Default**: ```30000, // 30 seconds```
+     *
+     * @type {number|undefined}
+     */
+    this.maxTransactionRetryTime = 30000 // 30 seconds
+
+    /**
+     * Specify socket connection timeout in milliseconds.
+     *
+     * Negative and zero values result in no timeout being applied.
+     * Connection establishment will be then bound by the timeout configured
+     * on the operating system level.
+     *
+     * **Default**: ```30000, // 30 seconds```
+     *
+     * @type {number|undefined}
+     */
+    this.connectionTimeout = 30000 // 30 seconds
+
+    /**
+     * Make this driver always return native JavaScript numbers for integer values, instead of the
+     * dedicated {@link Integer} class.
+     *
+     * Values that do not fit in native number bit range will be represented as `Number.NEGATIVE_INFINITY` or `Number.POSITIVE_INFINITY`.
+     *
+     * **Warning:** {@link ResultSummary} It is not always safe to enable this setting when JavaScript applications are not the only ones
+     * interacting with the database. Stored numbers might in such case be not representable by native
+     * `Number` type and thus driver will return lossy values. This might also happen when data was
+     * initially imported using neo4j import tool and contained numbers larger than
+     * `Number.MAX_SAFE_INTEGER`. Driver will then return positive infinity, which is lossy.
+     *
+     * **Default**: ```false```
+     *
+     * Default value for this option is `false` because native JavaScript numbers might result
+     * in loss of precision in the general case.
+     *
+     * @type {boolean|undefined}
+     */
+    this.disableLosslessIntegers = false
+
+    /**
+     * Make this driver always return native Javascript `BigInt` for integer values,
+     * instead of the dedicated {@link Integer} class or `Number`.
+     *
+     * **Warning:** `BigInt` doesn't implement the method `toJSON`. In manner of serialize it as `json`,
+     * this's needed to add a custom implementation of the `toJSON` on the
+     * `BigInt.prototype`. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#use_within_json.
```

Comment:
I don't understand the meaning of the sentence starting with `In manner` :thinking: 

Is this correct?

> In you want to serialize a `BigInt` value to json, you need a custom implementation of the method `toJSON` on `BigInt.prototype`.

Reply:
I add a suggestion in comment above